src/main/scala/mesosphere/marathon/api/serialization/ContainerSerializer.scala

```diff
@@ -22,57 +22,18 @@ object ContainerSerializer {
     val maybeDocker = if (proto.hasDocker) Some(DockerSerializer.fromProto(proto.getDocker)) else None
     Container(
       `type` = proto.getType,
-      volumes = proto.getVolumesList.asScala.map(Volume(_)).to[Seq],
+      volumes = proto.getVolumesList.asScala.map(Volume.fromProto(_)).to[Seq],
       docker = maybeDocker
     )
   }
 
   def toMesos(container: Container): mesos.Protos.ContainerInfo = {
-    // we can only serialize DockerVolumes into a Mesos Protobuf.
-    // PersistentVolumes and ExternalVolumes are handled differently
-    val serializedVolumes = container.volumes.collect { case dv: DockerVolume => VolumeSerializer.toMesos(dv) }
-    val builder = mesos.Protos.ContainerInfo.newBuilder
-      .setType(container.`type`)
-      .addAllVolumes(serializedVolumes.asJava)
+    val builder = mesos.Protos.ContainerInfo.newBuilder.setType(container.`type`)
     container.docker.foreach { d => builder.setDocker(DockerSerializer.toMesos(d)) }
     builder.build
   }
 }
 
-object VolumeSerializer {
```

Comment:
I was trying to be consistent w/ respect to how and where VolumesModule.inject was being invoked. I'll think about a better way to approach this. Moving the serialization to the objects felt natural because validation is also closely coupled w/ the business objects. In the volumes case, both aspects (serialization and validation) depend on the provider so ultimately I'd like to pull everything out of the "state" layer and into somewhere else.